test/sharness/t0184-http-proxy-over-p2p.sh

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env bash
+
+test_description="Test http proxy over p2p"
+
+. lib/test-lib.sh
+WEB_SERVE_PORT=5099
+IPFS_GATEWAY_PORT=5199
+SENDER_GATEWAY="http://127.0.0.1:$IPFS_GATEWAY_PORT"
+
+function show_logs() {
+
+    echo "*****************"
+    echo "  RECEIVER LOG  "
+    echo "*****************"
+    iptb logs 1
+    echo "*****************"
+    echo "  SENDER LOG  "
+    echo "*****************"
+    iptb logs 0
+    echo "*****************"
+    echo "REMOTE_SERVER LOG"
+    echo $REMOTE_SERVER_LOG
+    echo "*****************"
+    cat $REMOTE_SERVER_LOG
+}
+
+function serve_http_once() {
+    #
+    # one shot http server (via nc) with static body
+    #
+    local body=$1
+    local status_code=${2:-"200 OK"}
+    local length=$((1 + ${#body}))
+    REMOTE_SERVER_LOG="server.log"
+    echo -e "HTTP/1.1 $status_code\nContent-length: $length\n\n$body" | nc -l $WEB_SERVE_PORT 2>&1 > $REMOTE_SERVER_LOG &
```

Comment:
This is going to be horribly racy, I'd add:
```suggestion
    rm $REMOTE_SERVER_LOG
    echo -e "HTTP/1.1 $status_code\nContent-length: $length\n\n$body" | nc -l $WEB_SERVE_PORT 2>&1 > $REMOTE_SERVER_LOG &
    test_wait_for_file 30 100ms $REMOTE_SERVER_LOG
```